code/modules/fissionreactor/objects_external.dm

```diff
@@ -0,0 +1,736 @@
+			newframe.components+=/obj/item/weapon/stock_parts/matter_bin
+			newframe.components+=/obj/item/weapon/stock_parts/scanning_module
+		return
+				
+				
+				
+	if(associated_reactor && associated_reactor.considered_on())
+		return
+
+/obj/machinery/fissioncontroller/attack_hand(mob/user)
+	if(..())
+		return
+	if(!associated_reactor)
+		associated_reactor=new /datum/fission_reactor_holder
+		var/constructionerror=associated_reactor.init_resize(src.loc)
+		if(constructionerror)
+			say("Failed to setup reactor: [constructionerror]", class = "binaryradio")
+			associated_reactor=null
+			return
+		associated_reactor.init_parts()
+		associated_reactor.controller=src
+		if(currentfuelrod)
+			associated_reactor.fuel=currentfuelrod.fueldata
+		say("Reactor setup success.", class = "binaryradio")
+		update_icon()
+	
+
+/obj/machinery/fissioncontroller/update_icon()
+	if(!powered())
+		icon_state="control0"
+		return
+	if(stat & BROKEN)
+		icon_state="controlb"
+		return
+	if(!associated_reactor)
+		icon_state="control_noreactor"
+		return
+	if(!associated_reactor.fuel)
+		icon_state="control_nofuel"
+		return
+	if(associated_reactor.fuel.life <=0)
+		icon_state="control_depleted"
+		return
+	if(associated_reactor.temperature>=FISSIONREACTOR_DANGERTEMP)
+		icon_state="control_danger"
+		return
+	if(!associated_reactor.considered_on())
+		icon_state="control_idle"
+		return
+	icon_state="control"
+
+/obj/machinery/fissioncontroller/examine()
+	..()
+	to_chat(usr, "It's held together tightly, you'll have to cut the metal to take it apart.")
+	switch(icon_state)
+		if("control0")
+			to_chat(usr, "The power is off. You should plug it in. Soon.")
+			return
+		if("controlb")
+			to_chat(usr, "The screen is broken. You should fix it soon.")
+			return
+		if("control_noreactor")
+			to_chat(usr, "The readouts indicate there's no linked reactor.")
+		if("control_nofuel")
+			to_chat(usr, "The readouts indicate there's no fuel rod inserted.")
+		if("control_depleted")
+			to_chat(usr, "The readouts indicate that the fuel is depleted.")
+		if("control")
+			to_chat(usr, "The readouts indicate that the reactor is operating normally.")
+		if("control_idle")
+			to_chat(usr, "The readouts indicate that the reactor is shut down.")
+		if("control_danger")
+			to_chat(usr, "The readouts indicate that the reactor is overheated, and that you should cool it down.")
+		
+	to_chat(usr, "The temperature reads out [associated_reactor.temperature]K")
+	if(associated_reactor.fuel)
+		to_chat(usr, "The fuel reads out [floor(associated_reactor.fuel.life*100+0.5)]% life remaining")
+
+/*
+/obj/machinery/fissioncontroller/set_broken()
+	if(can_autoscram)
+		associated_reactor.SCRAM=TRUE
+		say("Reactor controller electrical fault detected, engaging SCRAM.", class = "binaryradio")
+		return TRUE
+	return FALSE
+*/
+
+/obj/machinery/fissioncontroller/process()
+	update_icon()
+	if(!associated_reactor) //no reactor? no processing to be done.
+		return	
```

Comment:
Remove it from the processing reactors list so it stops ticking

Reply:
this is ticking regardless of a reactor, eg, if you build the controller first before the whole reactor is assembled